README.md

```diff
@@ -1,31 +1,125 @@
+
+If you want to run Space Daemon by source, check out [this section](https://github.com/FleekHQ/space-daemon#Running)
+
+## Usage
+
+Space Daemon provides a gRPC interface. You can read its proto schema [here](https://github.com/FleekHQ/space-daemon/blob/master/grpc/pb/space.proto). It contains methods to:
+
+- Create files and directories
+
+- List files and directories
+
+- Creating buckets
+
+- Sharing buckets
+
+- Creating identities
+
+You can also use the JavaScript client here [https://github.com/FleekHQ/space-client](https://github.com/FleekHQ/space-client)
+
+This can be useful if, for example, you are building a web app that needs to interact with a user's locally running Space Daemon.
+
+## Modules
+
+Space Daemon requires a few modules to run successfully. If you downloaded the binary, you don't have to worry about this since it will be connecting to our services. It's good to understand what's happening behind the scenes though.
+
+### IPFS Node
+
+All encrypted files are stored in an IPFS node. For convenience, Space Daemon connects to a hosted IPFS node by default. You can connect to one of your choosing by providing the `-ipfsaddr` flag (e.g. `-ipfsaddr=/ip4/127.0.0.1/tcp/5001`)
+
+### Textile Hub
+
+Required for sharing files between users and backing it up. It stores all backed up files encrypted using a set of keys so that only you, and people you share files with, can read the data. We host our own instance of the Textile Hub, and by default, Space Daemon will conect to it. It can be customized by providing the `-textilehub` flag and `-textilethreads` flag.
+
+### Space Services
+
+We provide hosted alternatives for these services. You can deploy your own by following the instructions in its repo:
+
+[https://github.com/fleekHQ/space-services](https://github.com/fleekHQ/space-services)
+
+#### Identity
+
+These are centralized services that are optional, but offer additional convenience. Used mainly for identity. By using these services, you can allow users to claim usernames, so that Space Daemon can know the public key of a given username and in that way share files via username without having to input public keys directly.
+
+#### Authentication
+
+Our hosted Textile Hub requires authentication via public key for logging in. This service sends a challenge to Space Daemon, which signs the challenge with the private key of the user and in that way our hosted Textile Hub can allow the user to store data.
+
+### MongoDB
+
+Currently, local Textile Threads require a running MongoDB database. Space Daemon by default connects to a hosted one, but this will be removed once Textile switches over to an embeddable data store.
+
+## Running from source
+
+After cloning this repo, you can run it from source by running `go run ./cmd/space-daemon -devMode=true`. Consider that you will need the following environment variables exported in your system:
+
+```
+SPACE_APP_DIR=[The path where the source code is located]
```

Comment:
Actually we don't use that env anymore, since we are running config from `map_config` and there's no need to find the `space.json` anymore (which is what that env was used for). So I removed it in 7f689d3 and also removed that env from .env.example